### ISSUE TITLE
feature: Add a 'prepare' step as part of creation of VM

### ISSUE THREAD

**Issue by mzsilak** [metacloud/molecule#963]
# Issue Type
- Feature request

# Molecule and Ansible details

```
ansible 2.3.2.0
molecule master
```

- Molecule installation method: pip
- Ansible installation method: OS package

# Desired Behaviour

The _create_ step of a scenario shall be able to execute tasks on a VM after its creation through a new template file _prepare.yml_. The file shall tasks to prepare the VM for the intended scenario and role.

Minimum content (moved from playbook.yml):
```
- name: Prepare
  hosts: all
  gather_facts: False
  tasks:
    - name: Install python for Ansible
      raw: test -e /usr/bin/python || (yum clean all -y && yum install python-minimal -y)
      become: True
```

The _prepare.yml_ shall only be executed as part of VM creation. If the VM exists already it must not be executed to keep the idempotence of the _converge_ step

# Motivation

The current execution steps always target a fresh installation of a role. The addition of a _prepare_ step allows testing of roles in existing or (manually) modified environments and fits the possibility to create multiple scenarios with molecule.

**Comment by retr0h** [comment]
I'm planning to implement anything like this at moment.  The playbook can be used to run any prepare statements necessary.  If you are concerned about idempotence mark the tasks or roles running as `changed_when: False` in playbook.yml.  If the prepare work is done correctly it can be made idempotent so it does not impact the converge steps.

**Comment by mzsilak** [comment]
I guess you meant '... **not** planning to implement...'.

I have the issue that the my prepare tasks basically contradict the tasks in the role. So whenever I run the prepare tasks, the role has to change it back. (i.e. testing a package upgrade).

I created a fork and have some success already. I am not struggling with the tests and tox-tags. The changes I did so far would create a prepare.yml on new scenarios, but will also work with existing scenarios. If its fine for you, I would submit a pull request as soon as the tests are green.
